text/0000-cfg-oneof.md

```diff
@@ -0,0 +1,96 @@
+- Feature Name: `config_oneof`
+- Start Date: 2020-07-18
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Addition of `oneof` to existing `all`, `any`, `not` configuration predicates in `#[cfg!()]` macros
+
+# Motivation
+[motivation]: #motivation
+
+In a number of situations (particularly involving `no_std` and cross-target applications) it is useful to ensure that only one of a set of features are enabled.
+It is important to enforce this due to the additive behaviour of features (ie. dependencies may include the same sub-dependencie with different features enabled the result is a union of the enabled features), and difficult to specify for larger feature sets using existing predicates as the complexity for this increases exponentially with the number of exclusive features.
+
+The desired outcome of this is the ability to specify `#[cfg(oneof(feature="a", feature="b"))] do_something()` and `#[cfg(not(oneof(feature="a", feature="b")))] compile_error!(...)` to specify exclusive features without manually defining all possible valid/invalid combinations for the exclusive subset of features, and simplify other configurations that benefit from the `oneof` predicate.
+This allows authors of crates with exclusive feature sets to communicate this to consumers without requiring users to infer the feature issue from compiler errors.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Additions to existing [guide](https://doc.rust-lang.org/reference/conditional-compilation.html)
+
+``` 
+ConfigurationOneof
+   oneof ( ConfigurationPredicateList )
+   
+...
+
+oneof() with a comma separated list of configuration predicates. It is true if at one and only one predicate is true. In all other situations it is false.
+```
+
+Using an existing implementation from [rust-rand-facade](https://github.com/ryankurte/rust-rand-facade/blob/master/src/lib.rs) expressing all possible combinations for a set of three exclusive features (`std`, `cortex_m`, and `os_rng`).
```

Comment:
I don't understand what this sentence is trying to say. Is this still a part of the addition to the guide?

Reply:
I can't quite tell which sentence this comment refers to.

`oneof() with a comma separated...` in the source block is part of the guide, matching the existing predicate definitions.

`Using an existing implementation...` is intended as an example, I can clarify this

Reply:
> I can't quite tell which sentence this comment refers to.

The comment is placed on a line containing a single sentence and that's what the comment is about, so I'm not sure how I could have better communicated that.